lib/widget/survey_shimmer_loading.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+import 'package:shimmer/shimmer.dart';
+import 'package:survey_flutter_ic/theme/dimens.dart';
+import 'package:survey_flutter_ic/widget/place_holders.dart';
+
+class SurveyShimmerLoading extends StatelessWidget {
+  const SurveyShimmerLoading({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Shimmer.fromColors(
+        baseColor: Colors.grey.shade500,
+        highlightColor: Colors.grey.shade50,
+        enabled: true,
+        child: SafeArea(
+          child: Padding(
+              padding:
+                  const EdgeInsets.symmetric(horizontal: space20, vertical: 62),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  _buildHeaderShimmer(),
+                  const Expanded(child: SizedBox.shrink()),
+                  _buildPagerIndicatorPlaceholder(),
+                  _buildSurveyContent()
+                ],
+              )),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildHeaderShimmer() {
+    return Row(
+      crossAxisAlignment: CrossAxisAlignment.end,
+      children: [
+        Expanded(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.end,
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: const [
+              TextPlaceholder(width: 117.0),
+              SizedBox(height: space4),
+              TextPlaceholder(width: 97.0),
+            ],
+          ),
+        ),
+        const CirclePlaceholder(),
+      ],
+    );
+  }
+
+  Widget _buildPagerIndicatorPlaceholder() {
+    return const TextPlaceholder(width: 50);
+  }
+
+  Widget _buildSurveyContent() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: const [
+        SizedBox(height: space20),
+        TextPlaceholder(width: 253),
+        SizedBox(height: space4),
+        TextPlaceholder(width: 143),
+        SizedBox(height: space16),
+        TextPlaceholder(width: 318),
+        SizedBox(height: space4),
+        TextPlaceholder(width: 278)
```

Comment:
we can face the OverFlowException if the width is longer than the screen width of the small device. Can we set the width by using the percentage of the screen width? For example, `MediaQuery.of(context).size.width * 0.8` (take 80% of the screen width)